go.mod

```
@@ -1,6 +1,8 @@
 module github.com/ignite/cli
 
-go 1.20
+go 1.21.1
```

This version is the minimum version that packages importing `ignite` should use.
I would recommend to leave it as `1.21` or to be consistent with actual Go versions downgrade it to `1.20.0`:

```suggestion
go 1.21.0
```

The `toolchain` in the other hand could be updated to `go1.21.3` because is the one that ignite uses to compile the dependencies and binary I believe. If in doubt it would be safe to stay as `go1.21.0`.

---

After Go introduced the optional `toolchain` line I think there is no consensus on where to settle for the CLI. Maybe we could set `go` to the minimum Go version required to use the CLI packages, which means setting the lowest Go version that would compile and then use the `toolchain` to a recent Go version, which would require to also update the CI configurations.

Maybe in this PR we can settle to these values and **only** keep the `toolchain` and CI configuration "up to date":

```
go 1.21.0

toolchain go1.21.3
```

We would update the `go` line only when the CLI introduces a new feature that is not supported by `1.21.0`.

@Pantani @julienrbrt